source/common/router/config_impl.cc

```diff
@@ -539,6 +544,10 @@ RouteEntryImplBase::WeightedClusterEntry::WeightedClusterEntry(
       }
     }
   }
+
+  if (cluster.has_cluster_metadata()) {
+    cluster_metadata_.MergeFrom(cluster.cluster_metadata());
+  }
```

Comment:
How will this work when both have same keys?

Reply:
The docs for MergeFrom state that 'Singular fields will be overwritten, if specified in from, except for embedded messages which will be merged. Repeated fields will be concatenated'.
Are you asking specifically how MergeFrom works for maps?